tests/Tokenizer/Analyzer/ClassesAnalyzerTest.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Tokenizer\Analyzer;
+
+use PhpCsFixer\Tests\TestCase;
+use PhpCsFixer\Tokenizer\Analyzer\ClassesAnalyzer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Pol Dellaiera <pol.dellaiera@protonmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Tokenizer\Analyzer\ClassesAnalyzer
+ */
+final class ClassesAnalyzerTest extends TestCase
+{
+    /**
+     * @param int         $classIndex
+     * @param string      $code
+     * @param null|string $name
+     * @param null|string $startIndex
+     * @param null|string $endIndex
+     *
+     * @dataProvider provideGetExtendsClassCases
+     */
+    public function testGetClassExtends($classIndex, $code, $name = null, $startIndex = null, $endIndex = null)
+    {
+        $tokens = Tokens::fromCode($code);
+        $analyzer = new ClassesAnalyzer();
+
+        $extend = $analyzer->getClassExtends($tokens, $classIndex);
+
+        if (null !== $extend) {
+            static::assertSame($name, $extend->getName());
+            static::assertSame($startIndex, $extend->getStartIndex());
+            static::assertSame($endIndex, $extend->getEndIndex());
+        } else {
+            static::assertNull($extend);
+        }
+    }
+
+    public function provideGetExtendsClassCases()
+    {
+        return [
```

Comment:
Can we have test case(s) where `$classIndex` points to non-class, e.g. open tag and some `foreach` (that will have `{` as next token)?

Reply:
Fixed.